src/ILCompiler.CppCodeGen/src/CppCodeGen/ILToCppImporter.cs

```diff
@@ -2876,6 +2876,11 @@ private void ImportStoreField(int token, bool isStatic)
             Append(" = ");
             if (!fieldType.IsValueType)
             {
+                if (runtimeDeterminedOwningType.IsRuntimeDeterminedSubtype)
+                {
+                    fieldType = owningType.GetField(field.Name).FieldType;
```

Comment:
@MichalStrehovsky What would be the right type system API to call here, to avoid roundtrip via name?

Reply:
Should be:

```suggestion
                    fieldType = _typeSystemContext.GetFieldForInstantiatedType(field.GetTypicalFieldDefinition(), (InstantiatedType)owningType).FieldType;
```

It's  a bit more wordy, but much faster.